src/Language/PureScript/CodeGen/JS/Optimizer/Inliner.hs

```diff
@@ -233,6 +224,21 @@ inlineCommonOperators = applyAll $
     go m acc (JSApp _ lhs [arg]) = go (m - 1) (arg : acc) lhs
     go _ _   _ = Nothing
 
+  inlineNonClassFunction :: (JS -> Bool) -> (JS -> JS -> JS) -> JS -> JS
+  inlineNonClassFunction p f = everywhereOnJS convert
+    where
+    convert :: JS -> JS
+    convert (JSApp _ (JSApp _ op' [x]) [y]) | p op' = f x y
+    convert other = other
+
+  isModFn :: (String, String) -> JS -> Bool
+  isModFn (m, op) (JSAccessor _ op' (JSVar _ m')) = m == m' && op == op'
+  isModFn _ _ = False
+
+  isDictModFn :: (String, String, String) -> JS -> Bool
+  isDictModFn (d, m, op) (JSApp _ (JSAccessor _ op' (JSVar _ m')) [(JSVar _ d')]) = m == m' && op == op' && d' == "dict" ++ d
```

Comment:
I would ignore the dictionary name here entirely, actually. After type-checking, it's not like we can apply this function to something which is not a `Partial` dictionary anyway, so I would just leave it as a wildcard in case the code generation changes later.